pkg/stringset/set.go

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+package stringset
+
+import (
+	cuckoo "github.com/seiflotfy/cuckoofilter"
+)
+
+type Set interface {
+	Has(string) bool
+	HasAny([]string) bool
+}
+
+type fixedSet struct {
+	cuckoo *cuckoo.Filter
+}
+
+func (f fixedSet) HasAny(strings []string) bool {
+	for _, s := range strings {
+		if f.Has(s) {
+			return true
+		}
+	}
+	return false
+}
+
+func NewFromStrings(items ...string) Set {
+	f := cuckoo.NewFilter(uint(len(items)))
+	for _, label := range items {
+		f.InsertUnique([]byte(label))
+	}
+
+	return &fixedSet{cuckoo: f}
```

Comment:
Could we reuse the filters between updates using https://github.com/seiflotfy/cuckoofilter/blob/master/cuckoofilter.go#L45?

Reply:
Hmm.. yeah, good point. Theoretically, we could hold a lock during the period of "Reset->Add items" but then a StoreAPI would be unresponsive during that time. It's hard to tell but it's probably faster to allocate some new memory every time :sweat_smile: